src/Tizen.NUI/src/public/BaseComponents/Style/TextEditorStyle.cs

```diff
@@ -0,0 +1,523 @@
+            defaultValueCreator: (bindable) => ((TextEditorStyle)bindable).enableSelection);
+
+        /// <summary> The bindable property of MatchSystemLanguageDirection. </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        internal static readonly BindableProperty MatchSystemLanguageDirectionProperty = BindableProperty.Create(nameof(MatchSystemLanguageDirection), typeof(bool?), typeof(TextEditorStyle), null,
+            propertyChanged: (bindable, oldValue, newValue) => ((TextEditorStyle)bindable).matchSystemLanguageDirection = (bool?)newValue,
+            defaultValueCreator: (bindable) => ((TextEditorStyle)bindable).matchSystemLanguageDirection);
+
+        private HorizontalAlignment? horizontalAlignment;
+        private Vector4 secondaryCursorColor;
+        private bool? enableCursorBlink;
+        private float? cursorBlinkInterval;
+        private float? cursorBlinkDuration;
+        private int? cursorWidth;
+        private string grabHandleImage;
+        private string grabHandlePressedImage;
+        private float? scrollThreshold;
+        private float? scrollSpeed;
+        private Vector4 selectionHighlightColor;
+        private Rectangle decorationBoundingBox;
+        private Vector4 inputColor;
+        private string inputFontFamily;
+        private float? inputPointSize;
+        private string inputUnderline;
+        private string inputShadow;
+        private string emboss;
+        private string inputEmboss;
+        private string inputOutline;
+        private float? pixelSize;
+        private bool? enableSelection;
+        private bool? matchSystemLanguageDirection;
+        private string fontFamily;
+        private Vector4 textColor;
+        private float? pointSize;
+        private Color placeholderTextColor;
+        private Vector4 primaryCursorColor;
+
+        static TextEditorStyle() { }
+
+        /// <summary>
+        /// Create an empty instance.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public TextEditorStyle() : base()
+        {
+        }
+
+        /// <summary>
+        /// The FontFamily property.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public string FontFamily
+        {
+            get => (string)GetValue(FontFamilyProperty);
+            set => SetValue(FontFamilyProperty, value);
+        }
+
+        /// <summary>
+        /// The HorizontalAlignment property.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public HorizontalAlignment? HorizontalAlignment
+        {
+            get => (HorizontalAlignment?)GetValue(HorizontalAlignmentProperty);
+            set => SetValue(HorizontalAlignmentProperty, value);
+        }
+
+        /// <summary>
+        /// The SecondaryCursorColor property.
+        /// Note that the cascade chaining set is impossible. Please set a whole value.
```

Comment:
Just my curiosity :
It means user should set  the  whole value of `Vector4` type, right?

Reply:
Yes right :) Normally the style does not need to edit once it created.